File: TP2DPBO2025C2/Java/Kode Program/Main.java
import java.util.ArrayList;
import java.util.Scanner;


public class Main {

    /**
     * @param args
     */
    public static void main(String[] args) {

        ArrayList<Baju> Data_Baju = new ArrayList<>();

        Data_Baju.add(new Baju("B001", "Kaos", "50000", 10, "Casual", "Katun", "Biru", "Pria", "M", "Nike"));
        Data_Baju.add(new Baju("B002", "Kemeja", "75000", 15, "Formal", "Poliester", "Putih", "Wanita", "L", "Zara"));
        Data_Baju.add(new Baju("B003", "Jaket", "120000", 7, "Casual", "Denim", "Hitam", "Unisex", "XL", "Levi's"));
        Data_Baju.add(new Baju("B004", "Sweater", "90000", 12, "Casual", "Wol", "Abu-abu", "Pria", "L", "Adidas"));
        Data_Baju.add(new Baju("B005", "Blouse", "85000", 20, "Formal", "Sutra", "Merah", "Wanita", "M", "Gucci"));

        String Pilihan;
        Scanner Scan = new Scanner(System.in);


        do{
            System.out.println();
            System.out.println("Input 1 untuk menambah data baru");
            System.out.println("Input 2 untuk print");
            System.out.println("Input 3 untuk keluar program");

            System.out.print("Masukkan inputan Pilihan produk yang ingin dicarii: ");
            Pilihan = Scan.next();

            if (Pilihan.equals("1")) {

                Baju Data_Temp = new Baju();

                System.out.print("Masukkan Pilihan ID baru : ");
                Data_Temp.setid(Scan.next());
    
                System.out.print("Masukkan Nama baru : ");
                Data_Temp.setnama_produk(Scan.next());
    
                System.out.print("Masukkan harga baru : ");
                Data_Temp.setharga_produk(Scan.next());
    
                System.out.print("Masukkan stok baru : ");
                Data_Temp.setstok_produk(Scan.nextInt());  

                System.out.print("Masukkan Pilihan Jenis baru : ");
                Data_Temp.set_jenis(Scan.next());
    
                System.out.print("Masukkan Bahan baru : ");
                Data_Temp.set_bahan(Scan.next());
    
                System.out.print("Masukkan Warna baru : ");
                Data_Temp.set_warna(Scan.next());

                System.out.print("Masukkan untuk baru : ");
                Data_Temp.set_untuk(Scan.next());

                System.out.print("Masukkan size baru : ");
                Data_Temp.set_size(Scan.next());

                System.out.print("Masukkan merk baru : ");
                Data_Temp.set_merk(Scan.next());     
                
                Data_Baju.add(Data_Temp);
                
            }else if (Pilihan.equals("2")) { // Menambahkan data baru


                int id_terpanjang = 2, nama_terpanjang = 4, harga_terpanjang = 5, stok_terpanjang = 4;
                int jenis_terpanjang = 5, bahan_terpanjang = 5, warna_terpanjang = 5, untuk_terpanjang = 5;
                int size_terpanjang = 4, merk_terpanjang = 4;
                int panjang_line=0;

                // Cari panjang maksimum dari setiap atribut
                for (int i = 0; i < Data_Baju.size(); i++) {
                    Baju baju = Data_Baju.get(i); // Ambil objek baju dari list
                    
                    id_terpanjang = Math.max(id_terpanjang, baju.getid().length());
                    nama_terpanjang = Math.max(nama_terpanjang, baju.getnama_produk().length());
                    harga_terpanjang = Math.max(harga_terpanjang, baju.getharga_produk().length());
                    stok_terpanjang = Math.max(stok_terpanjang, String.valueOf(baju.getstok_produk()).length());
                    jenis_terpanjang = Math.max(jenis_terpanjang, baju.getJenis().length());
                    bahan_terpanjang = Math.max(bahan_terpanjang, baju.getBahan().length());
                    warna_terpanjang = Math.max(warna_terpanjang, baju.getWarna().length());
                    untuk_terpanjang = Math.max(untuk_terpanjang, baju.getUntuk().length());
                    size_terpanjang = Math.max(size_terpanjang, baju.getSize().length());
                    merk_terpanjang = Math.max(merk_terpanjang, baju.getMerk().length());

                    int temp = id_terpanjang +nama_terpanjang +harga_terpanjang+stok_terpanjang+jenis_terpanjang+bahan_terpanjang+warna_terpanjang+untuk_terpanjang+size_terpanjang+merk_terpanjang;

                    if(temp > panjang_line+11){
                        panjang_line = temp;
                    }

                }
        
                String garis = "-".repeat(panjang_line + 11 * 3 - 2);
                System.out.println(garis);

                // Cetak header tabel
                System.out.printf("| %-" + id_terpanjang + "s | %-" + nama_terpanjang + "s | %-" + harga_terpanjang + "s | %-" + stok_terpanjang + "s | %-" + 
                jenis_terpanjang + "s | %-" + bahan_terpanjang + "s | %-" + warna_terpanjang + "s | %-" + 
                untuk_terpanjang + "s | %-" + size_terpanjang + "s | %-" + merk_terpanjang + "s |\n",
                "ID", "Nama", "Harga", "Stok", "Jenis", "Bahan", "Warna", "Untuk", "Size", "Merk");

                for (int j = 0; j < Data_Baju.size(); j++) {
                    Baju baju = Data_Baju.get(j); // Ambil objek Baju dari list
                
                    System.out.printf("| %-" + id_terpanjang + "s | %-" + nama_terpanjang + "s | %-" + harga_terpanjang + "s | %-" + stok_terpanjang + "s | %-" + 
                                      jenis_terpanjang + "s | %-" + bahan_terpanjang + "s | %-" + warna_terpanjang + "s | %-" + 
                                      untuk_terpanjang + "s | %-" + size_terpanjang + "s | %-" + merk_terpanjang + "s |\n",
                                      baju.getid(), baju.getnama_produk(), baju.getharga_produk(), baju.getstok_produk(), 
                                      baju.getJenis(), baju.getBahan(), baju.getWarna(), baju.getUntuk(), 
                                      baju.getSize(), baju.getMerk());
                }
                
                System.out.println(garis);   

                
              
                               
            }else if (Pilihan.equals("3")) { // Exit

                System.out.println("Terimakasih sudah menggunakan Java Ajam");
            }else{

                System.out.println("Syntax anda Error!!!");
            }

        }while(!Pilihan.equals("3"));

        Scan.close();
        

    }
}

File: TP2DPBO2025C2/cpp/Kode Program/main.cpp
#include <iostream>
#include <vector>
#include "Baju.cpp"

using namespace std;

int main() {
    vector<Baju> Data_Baju;

    // Menambahkan data awal dengan objek sementara
    Baju baju1("B001", "Kaos", "50000", 10, "Casual", "Katun", "Biru", "Pria", "M", "Nike");
    Baju baju2("B002", "Kemeja", "75000", 15, "Formal", "Poliester", "Putih", "Wanita", "L", "Zara");
    Baju baju3("B003", "Jaket", "120000", 7, "Casual", "Denim", "Hitam", "Unisex", "XL", "Levi's");
    Baju baju4("B004", "Sweater", "90000", 12, "Casual", "Wol", "Abu-abu", "Pria", "L", "Adidas");
    Baju baju5("B005", "Blouse", "85000", 20, "Formal", "Sutra", "Merah", "Wanita", "M", "Gucci");

    // Menyimpan objek ke dalam vector
    Data_Baju.push_back(baju1);
    Data_Baju.push_back(baju2);
    Data_Baju.push_back(baju3);
    Data_Baju.push_back(baju4);
    Data_Baju.push_back(baju5);

    string Pilihan;
    do {
        cout << "\nInput 1 untuk menambah data baru" << endl;
        cout << "Input 2 untuk print" << endl;
        cout << "Input 3 untuk keluar program" << endl;
        cout << "Masukkan pilihan: ";
        cin >> Pilihan;

        if (Pilihan == "1") {
            Baju Data_Temp;
            string id, nama, harga, jenis, bahan, warna, untuk, size, merk;
            int stok;

            cout << "Masukkan ID baru: ";
            cin >> id;
            Data_Temp.set_id(id);

            cout << "Masukkan Nama baru: ";
            cin >> nama;
            Data_Temp.set_nama_produk(nama);

            cout << "Masukkan Harga baru: ";
            cin >> harga;
            Data_Temp.set_harga_produk(harga);

            cout << "Masukkan Stok baru: ";
            cin >> stok;
            Data_Temp.set_stok_produk(stok);

            cout << "Masukkan Jenis baru: ";
            cin >> jenis;
            Data_Temp.setJenis(jenis);

            cout << "Masukkan Bahan baru: ";
            cin >> bahan;
            Data_Temp.setBahan(bahan);

            cout << "Masukkan Warna baru: ";
            cin >> warna;
            Data_Temp.setWarna(warna);

            cout << "Masukkan Untuk: ";
            cin >> untuk;
            Data_Temp.setUntuk(untuk);

            cout << "Masukkan Size: ";
            cin >> size;
            Data_Temp.setSize(size);

            cout << "Masukkan Merk: ";
            cin >> merk;
            Data_Temp.setMerk(merk);

            Data_Baju.push_back(Data_Temp);

        } else if (Pilihan == "2") {
            cout << "---------------------------------------------" << endl;
            cout << "ID  | Nama   | Harga  | Stok | Jenis  | Bahan | Warna | Untuk | Size | Merk" << endl;
            cout << "---------------------------------------------" << endl;

            int id_terpanjang = 2, nama_terpanjang = 4, harga_terpanjang = 5, stok_terpanjang = 4;
            int jenis_terpanjang = 5, bahan_terpanjang = 5, warna_terpanjang = 5, untuk_terpanjang = 5;
            int size_terpanjang = 4, merk_terpanjang = 4;
            int panjang_line = 0;
        
            // Loop untuk mencari string terpanjang dari setiap atribut
            for (int i = 0; i < Data_Baju.size(); i++) {
                if (Data_Baju[i].get_id().length() > id_terpanjang)
                    id_terpanjang = Data_Baju[i].get_id().length();

                if (Data_Baju[i].get_nama_produk().length() > nama_terpanjang)
                    nama_terpanjang = Data_Baju[i].get_nama_produk().length();

                if (Data_Baju[i].get_nama_produk().length() > nama_terpanjang)
                    nama_terpanjang = Data_Baju[i].get_nama_produk().length();

                if (Data_Baju[i].get_harga_produk().length() > harga_terpanjang)
                    harga_terpanjang = Data_Baju[i].get_harga_produk().length();

                if (to_string(Data_Baju[i].get_stok_produk()).length() > stok_terpanjang)
                    stok_terpanjang = to_string(Data_Baju[i].get_stok_produk()).length();

                if (Data_Baju[i].getJenis().length() > bahan_terpanjang)
                    bahan_terpanjang = Data_Baju[i].getJenis().length();

                if (Data_Baju[i].getBahan().length() > bahan_terpanjang)
                    bahan_terpanjang = Data_Baju[i].getBahan().length();

                if (Data_Baju[i].getWarna().length() > warna_terpanjang)
                    warna_terpanjang = Data_Baju[i].getWarna().length();

                if (Data_Baju[i].getUntuk().length() > untuk_terpanjang)
                    untuk_terpanjang = Data_Baju[i].getUntuk().length();
                    
                if (Data_Baju[i].getSize().length() > size_terpanjang)
                    size_terpanjang = Data_Baju[i].getSize().length();
                    
                if (Data_Baju[i].getMerk().length() > merk_terpanjang)
                    merk_terpanjang = Data_Baju[i].getMerk().length();
            }

            panjang_line = id_terpanjang + nama_terpanjang +harga_terpanjang+ stok_terpanjang+ jenis_terpanjang + bahan_terpanjang + warna_terpanjang + untuk_terpanjang + size_terpanjang + merk_terpanjang;

           for (int i = 0; i < Data_Baju.size(); i++) {
                cout << Data_Baju[i].get_id() << " | "
                    << Data_Baju[i].get_nama_produk() << " | "
                    << Data_Baju[i].get_harga_produk() << " | "
                    << Data_Baju[i].get_stok_produk() << " | "
                    << Data_Baju[i].getJenis() << " | "
                    << Data_Baju[i].getBahan() << " | "
                    << Data_Baju[i].getWarna() << " | "
                    << Data_Baju[i].getUntuk() << " | "
                    << Data_Baju[i].getSize() << " | "
                    << Data_Baju[i].getMerk() << endl;
            }

            cout << "---------------------------------------------" << endl;

        } else if (Pilihan == "3") {
            cout << "Terima kasih sudah menggunakan program ini!" << endl;
        } else {
            cout << "Syntax anda Error!!!" << endl;
        }

    } while (Pilihan != "3");

    return 0;
}
